### ISSUE TITLE
Converted published Knora data in a second triple - store

### ISSUE THREAD

**Issue by janCstoffregen** [dhlab-basel/Knora#1436]
Hi guys,

we thought about the following approach, please let us know what you think about it. The goal is to copy published data from Knora to a second published triple - store and convert it there to data that is modelled in standardised ontologies.

1. All published Knora data is queried directly from the triple store and transferred to a second published triple store.
2. The ontologies of the not so ontology - layer intensive “exported published data” from Knora are mapped to external ontologies. 
    1. For example projectOne:creator is mapped to DC:creator
3. The mapping can be done with rules, we have a working proof of concept. No additional technological dependency is needed. Data modelled in one ontology is converted to another ontology.
4. The data in the openly accessible triple - store is accessible and easily combinable with external data that uses the same Ontologies, etc. This way the data becomes interoperable and reusable in an easy and standardised way.

We had this idea for quite some time, now would be a good moment again to have your feedback. It would allow us to use our generic ontologies in the published triple store and have everything in Knora in the way you want it to be modelled as well, we could combine your and our approach.

It was Hans idea, and we think it would be the way to go, it’s easily understandable and we could keep on writing generic ontologies and we can write the conversion rules as well.

In other words, this would mean no effort from your side, and it would be an add-on which is not affecting any existing functionality of Knora.

Thank you very much for your answers and have a nice week,
Jan

**Comment by tobiasschweizer** [comment]
> This way the data becomes interoperable and reusable in an easy and standardised way

i.e. a SPARQL endpoint?

**Comment by subotic** [comment]
AFAIK requests to the V2 API can also return Turtle. You could write a set of Gravsearch queries, that would return the complete project data. There are performance concerns though. It would probably make more sense, to create an export route, that would provide a download link to the data, then you could load this into your triplestore.

**Comment by lrosenth** [comment]
i can see this as a good way out of the "deadlock" and I'm very positive to it!

**Comment by janCstoffregen** [comment]
@tobiasschweizer yes, a Sparql - endpoint of an external second triple store what do you think about it?
@subotic we thought we copy everything once per night to the other triple store, so it does not need to be fast. An export opportunity of all the published data would be amazing though!

**Comment by janCstoffregen** [comment]
> i can see this as a good way out of the "deadlock" and I'm very positive to it!

I'm glad, I think this would be good way out as well!

**Comment by subotic** [comment]
We anyway need to provide the ability for export. We will have to see about the scheduling of this feature. For now, you have your own GraphDB instance, so you can do it easily yourself with a simple script. So we are not under time pressure to have this feature in production.

**Comment by janCstoffregen** [comment]
This is true, a fast implementation of the export feature is not necessary, I just wanted to know what you think about the idea of ontology conversion for published data, having it on a second triple store with a sparql endpoint.
